apps/speed.c

```diff
@@ -2713,6 +2774,48 @@ int speed_main(int argc, char **argv)
             print_result(D_CBC_CAST, testnum, count, d);
         }
     }
+#endif
+#ifndef OPENSSL_NO_SM4
+    if (doit[D_CBC_SM4]) {
+        EVP_CIPHER_CTX *ctx;
+        int outl, rc;
+
+        if (async_jobs > 0) {
```

Comment:
It's better to put the code into a function, like EVP_Digest_SM3_loop, then invoked by run_benchmark().

Reply:
I thought about implementing it like you said, because SM4 also needs to initialize `EVP_CIPHER_CTX`, unlike other algorithms, it cannot be simply called in run_benchmark().

Reply:
Fortunately this has been done in OpenSSL 3.0 ;-)